src/test/scala/chiselTests/naming/IdentifierSpec.scala

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/* TODO(azidar):
+ - (future PR) instance identifiers
+ - test non-case class arguments
+ - investigate definitionIdentifiers for more cornercases
+ */
+
+package chiselTests
+package naming
+
+import chisel3.{assert => _, _}
+import chisel3.experimental.BaseModule
+import chisel3.naming.{fixTraitIdentifier, IdentifierProposer}
+
+class ConcreteClass(i: Int) extends Module
+trait NormalTrait extends Module
+abstract class AbstractClass extends Module
+abstract class AbstArgsClass(i: Int) extends Module
+abstract class NestedAbtrClz(b: Int) extends AbstArgsClass(b)
+class NestedCrctClz(b: Int) extends ConcreteClass(b)
+class NestedTratClz(b: Int) extends NormalTrait
+@fixTraitIdentifier
+trait FixedNormalTrait extends Module
+
+class HasByNameArg(g: => Module) extends Module {
+  val x = Module(g)
+}
+
+class IdentifierSpec extends ChiselFunSpec with Utils {
+  it("(1): definitionIdentifier works on classes, abstract classes, but not traits") {
+    class TopA extends Module {
+      assert(Module(new ConcreteClass(0)).definitionIdentifier == "ConcreteClass_0")
+      assert(Module(new ConcreteClass(0) {}).definitionIdentifier == "ConcreteClass_0_1")
```

Comment:
what is the `_1` coming from? is this the auto-namespace-collision-avoidance? Can we just have an error if there is a namespace collision in the space of `definitionIdentifier`?

Reply:
We could but I'd prefer to do that in a future version, right now its not used anywhere, so I don't want to just cause additional errors.